src/MovingAverage.cpp

```diff
@@ -0,0 +1,111 @@
+/*
+ * Moving Average Filter (linear FIR filter with equal weights)
+ */
+#include "iarc7_sensors/MovingAverage.hpp"
+
+#include <ros/ros.h>
+
+namespace iarc7_sensors {
+
+MovingAverage::MovingAverage(const ros::Duration& time) : filter_time_constant_(time) {}
+
+void MovingAverage::updateFilter(double value, double derivative, const ros::Time& time) {
```

Comment:
Ok that's fine, it's good that the interface is there then. I assume you'll make a trello card.

No, I mean what you have now but using a generic data type for data points. That type could overload the operators used by this class to make calculations and then apply those operations to multiple data types that all have the same timestamps. So you could average position, velocity, and acceleration of one thing without rewriting this class.